jacodb-analysis/src/main/kotlin/org/jacodb/analysis/codegen/language/impl/JavaLanguage.kt

```diff
@@ -483,15 +486,60 @@ class JavaLanguage : TargetLanguage {
             }
         }
     }
-
-    override fun dumpFunction(func: FunctionPresentation, pathToSourcesDir: Path) = inFile(classNameForStaticFunction(func.shortName), pathToSourcesDir) {
-        appendStaticFunction(func)
+    private fun functionToTypeImpl(name: String = "", func: FunctionPresentation): TypeImpl {
+        val typeToReturn = TypeImpl(
+            shortName = classNameForStaticFunction(func.shortName),
+            defaultConstructorGraphId = func.graphId,
+            inheritanceModifier = InheritanceModifier.OPEN,
+        )
+        val funcToMethod = MethodImpl(
+            graphId = func.graphId,
+            containingType = typeToReturn,
+            name = func.shortName,
+            visibility = func.visibility,
+            returnType = func.returnType,
+            inheritanceModifier = InheritanceModifier.STATIC,
+            inheritedFrom = null,
+            parameters = func.parameters.map { Pair(it.usage, it.shortName) }
+        )
+        funcToMethod.visibleLocals.addAll(func.localVariables)
+        funcToMethod.callSites.addAll(func.callSites)
+        func.preparationSite.expressionsBefore.forEach { expBefore -> funcToMethod.preparationSite.addBefore(expBefore) }
+        func.preparationSite.expressionsAfter.forEach { expAfter -> funcToMethod.preparationSite.addAfter(expAfter) }
+        func.terminationSite.expressionsBefore.forEach { expBefore -> funcToMethod.terminationSite.addBefore(expBefore) }
+        func.terminationSite.expressionsAfter.forEach { expAfter -> funcToMethod.terminationSite.addAfter(expAfter) }
+        func.terminationSite.dereferences.forEach { deref -> funcToMethod.terminationSite.addDereference(deref) }
+        typeToReturn.typeParts.add(funcToMethod)
+        if (name.isNotEmpty()) {
+            typeToReturn.createMethod(
+                graphId = func.graphId,
+                name = "main",
+                inheritanceModifier = InheritanceModifier.STATIC,
+                parameters = listOf(
+                    Pair<TypeUsage, String>(
+                        InstanceTypeImpl(
+                            isNullable = false,
+                            typePresentation = TypeImpl(
+                                shortName = "String[]",
+                            ),
+                        ),
+                        "args"
+                    )
+                )
+            )
+            (typeToReturn.typeParts[1] as MethodImpl).preparationSite.addBefore(
```

Comment:
please dont do that

Reply:
Don't cast? Or don't add before in preparationSite explicitly?